### PR TITLE
Add basic hint functionality

### PR DESCRIPTION
**Pull request by agnesmardh**

### CODE REVIEW

**Comment by agnesmardh** [comment]
There are some things left that maybe should be fixed.
 The hint index is in the beginning -1 so the first hint is shown 
 There is a lot going on in Gameboard right now, some things can be own entities/components
 The hint index and help text indes could be in store instead of gameboard

File: src/components/GameBoard.vue [hunk 1]
```diff
@@ -4,17 +4,26 @@
     <h3>Bra jobbat!</h3>
     <p>Du klarade nivå {{completedLevel}}</p>
   </div>
-  <div class="image-container">
+  <div class="image-container" v-bind:class="{ 'image-container-hidden': duckHidden }">
```
Comment (williamleven) [inline src/components/GameBoard.vue]: `:class`  binds to `v-bind:class`

Nothing that has to be changed.

File: src/components/GameBoard.vue [hunk 2]
```diff
@@ -4,17 +4,26 @@
...
+  <div @click="closeSpeechBubble">
+    <vue-p5
+            @setup="setup"
+            @draw="draw"
+            @preload="preload"
+    ></vue-p5>
+  </div>
+  <button class="button hint-button" @click="getHint">Ledtråd</button>
+  <div v-if="speechBubble !== ''" class="box">
+    <p>{{speechBubble}}</p>
```
Comment (williamleven) [inline src/components/GameBoard.vue]: Neet with the computed property

File: src/components/GameBoard.vue [hunk 3]
```diff
@@ -41,6 +50,8 @@ export default {
         y: 0,
       },
       completedLevel: null,
+      hintIndex: -1, // So that we not skip the first hint the first time
```
Comment (williamleven) [inline src/components/GameBoard.vue]: This seems like an odd workaround

File: src/components/GameBoard.vue [hunk 4]
```diff
@@ -104,7 +131,7 @@ export default {
       // Cover 100 % of height
       sketch.createCanvas(sketch.windowHeight * 0.86, sketch.windowHeight)
       sketch.background(200)
-      this.duckPressed()
+      this.duckHidden = false
```
Comment (williamleven) [inline src/components/GameBoard.vue]: I don't think this should be needed as it should be false to begin with

File: src/components/GameBoard.vue [hunk 5]
```diff
@@ -124,30 +151,45 @@ export default {
     endTransition () {
       this.completedLevel = null
       this.duckHidden = false
-      this.duckPressed()
+      this.showHint = false
+      this.hintIndex = -1
```
Comment (williamleven) [inline src/components/GameBoard.vue]: See above comment about -1

File: src/store.js [hunk 6]
```diff
@@ -26,6 +26,7 @@ export default new Vuex.Store({
     level: 0, // The current level
     gameObjects: initialGameObjects,
     userFunctions: getFunctions(initialGameObjects, 0), // works as a cache
+    hint: 0, // The hint to be shown
```
Comment (williamleven) [inline src/store.js]: Seems to be unused

File: src/store.js [hunk 7]
```diff
@@ -72,6 +74,9 @@ export default new Vuex.Store({
     setEntities (state, entities) {
       Vue.set(state, 'entities', entities)
     },
+    incHint (state) {
```
Comment (williamleven) [inline src/store.js]: Seems to be unused

File: src/store.js [hunk 8]
```diff
@@ -35,6 +36,7 @@ export default new Vuex.Store({
     getCharacters: state => state.gameObjects.filter(Obj => (new Obj()) instanceof Character),
     getLevelObstacles: state => state.levels[state.level].obstacles,
     getLevel: state => state.level,
+    getHint: state => state.levels[state.level].hints,
```
Comment (williamleven) [inline src/store.js]: should probably be named `getHints`

**Review by williamleven — CHANGES_REQUESTED** [review]
I've commented on some minor problems of which some should be resolved before merge. 

I also think that the functionality of this component should be split up into some smaller components.

**Review by williamleven — APPROVED** [review]

**Pull request merged**